automation/validate-pvc-name-stability.py

```diff
@@ -0,0 +1,103 @@
+#! python
+
+import logging
+import os
+import os.path
+import yaml
+import subprocess
+
+
+def validatePVCNames(path, liveTemplates):
+    templates = [f for f in os.listdir(path) if os.path.isfile(os.path.join(path, f))]
+    errors = []
+    for templateFilename in templates:
+        with open(os.path.join(path, templateFilename), 'r') as stream:
+            try:
+                template = yaml.safe_load(stream)
+
+                if template == None:
+                    logging.info("Empty template file: %s", template)
+                    continue
+
+                templateName = template["metadata"]["name"]
+                logging.info("Checking PVC name stability for: {}".format(templateName))
+
+                pvcName = getPVCNameFrom(template)
+                pvcNamespace = getPVCNamespaceFrom(template)
+
+                matchingLiveTemplate = liveTemplates.get(templateName, None)
+                if matchingLiveTemplate:
+                    liveTemplatePVCName = getPVCNameFrom(matchingLiveTemplate)
+                    liveTemplatePVCNamespace = getPVCNamespaceFrom(matchingLiveTemplate)
+
+                    if pvcName != liveTemplatePVCName:
+                        errors.append(pvcNameErrorMessage(pvcName, templateName))
+                    if pvcNamespace != liveTemplatePVCNamespace:
+                        errors.append(pvcNamespaceErrorMessage(pvcNamespace, templateName))
+                else:
+                    logging.info("Missing liveTemplate for {}".format(templateName))
+
+            except yaml.YAMLError as exc:
+                raise exc
+            except Exception as exc:
+                logging.error(exc)
+
+    if len(errors) > 0:
+        error_message = "\n".join(errors)
+        logging.info(FAILED_INFO_MESSAGE)
+        raise Exception(error_message)
+
+
+def getParamFrom(template, paramName):
+    for param in template["parameters"]:
+        if param["name"] == paramName:
+            return param["value"]
+
+
+def getPVCNameFrom(template):
+    return getParamFrom(template, "SRC_PVC_NAME")
+
+
+def getPVCNamespaceFrom(template):
+    return getParamFrom(template, "SRC_PVC_NAMESPACE")
+
+
+def fetchLiveTemplates():
+    out = subprocess.run(["oc", "get", "template", "-o", "yaml"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
what about using python openshift package? https://github.com/openshift/openshift-restclient-python

Reply:
Oh, cool. I didn't know that exists.
I thought about adding the kubernetes client and using that, but decided to avoid any additional dependencies for the test.
Didn't want to add any python requirements as I didn't see any managed right now. But I can use the client if you'd prefer that.

Reply:
We already have python dependencies (see https://github.com/kubevirt/common-templates/blob/master/builder/Dockerfile#L4). Please if you have time try to use the python client.

Reply:
Done, thanks for pointing me in the right direction.